src/main/java/org/folio/linked/data/mapper/dto/monograph/work/sub/AgentMapperUnit.java

```diff
@@ -34,9 +31,8 @@ public <P> P toDto(Resource source, P parentDto, Resource parentResource) {
   @Override
   public Resource toEntity(Object dto, Resource parentEntity) {
     var agent = (Agent) dto;
-    var resource = resourceRepository.findById(Long.parseLong(agent.getId()))
-      .map(Resource::copyWithNoEdges)
-      .orElseThrow(() -> new NotFoundException(RESOURCE_WITH_GIVEN_ID + agent.getId() + IS_NOT_FOUND));
+    var resource = fetchResource(agent, idVal -> resourceRepository.findById(Long.parseLong(idVal)),
```

Comment:
Is there any advantage of passing the same functions from here & from [ReferenceMapperUnit.java](https://github.com/FOLIO-EIS/mod-linked-data/pull/358/files#diff-c716844f3023809d8475e287ab4922da7ebea94239337305e47b0285a7e1f1f8) to `fetchResource` method?


May be we can simplify like follows?

var resource = fetchResource(agent);

.. and 

public interface WorkSubResourceMapperUnit extends ... {
      default Resource fetchResource(Identifiable identifiable) {
           var fetchByIdFunction = idVal -> resourceRepository.findById(Long.parseLong(idVal);
           var fetchBySrsIdFunction = resourceRepository::findByFolioMetadataSrsId;
           
          // existing logic.
          
      } 
}

Reply:
If I'm not mistaken, we cannot do exactly like this because ```interface``` can't have state.
As an option we can pass ```ResourceRepository resourceRepository``` to the ```fetchResource``` method.